Code/Source/VehicleDynamics/DriveModels/SimplifiedDriveModel.cpp

```diff
@@ -0,0 +1,98 @@
+    }
+
+    void SimplifiedDriveModel::ApplyInputState(const VehicleInputsState& inputs, const ChassisConfiguration& vehicleChassis, uint64_t nsDt)
+    {
+        ApplySteering(inputs.m_steering, vehicleChassis, nsDt);
+        ApplySpeed(inputs.m_speed, vehicleChassis, nsDt);
+    }
+
+    // TODO - speed and steering handling is quite similar, possible to refactor?
+    void SimplifiedDriveModel::ApplySteering(float steering, const ChassisConfiguration& vehicleChassis, uint64_t nsDt)
+    {
+        auto steeringEntities = VehicleDynamics::Utilities::GetAllSteeringEntities(vehicleChassis);
+        for (auto& steeringEntity : steeringEntities)
+        {
+            AZ::Vector3 currentSteeringElementRotation;
+            AZ::TransformBus::EventResult(currentSteeringElementRotation, steeringEntity, &AZ::TransformBus::Events::GetLocalRotation);
+            auto currentSteeringAngle = currentSteeringElementRotation.GetZ(); // TODO - axis could be different
+            double pidCommand = m_steeringPid.computeCommand(steering - currentSteeringAngle, nsDt);
+            AZ_TracePrintf(
+                "SimplifiedDriveModel",
+                "Applying steering to entity %s with values: desired %f, current %f, impulse %lf\n",
+                steeringEntity.ToString().c_str(),
+                steering,
+                currentSteeringAngle,
+                pidCommand);
+
+            if (AZ::IsClose(pidCommand, 0.0)) // TODO - use the third argument with some reasonable value which means "close enough"
+            {
+                continue;
+            }
+
+            auto torque = pidCommand;
+
+            AZ::Transform steeringElementTransform;
+            AZ::TransformBus::EventResult(steeringElementTransform, steeringEntity, &AZ::TransformBus::Events::GetWorldTM);
+            auto transformedTorqueVector = steeringElementTransform.TransformVector(AZ::Vector3(0, 0, torque));
+            Physics::RigidBodyRequestBus::Event(steeringEntity, &Physics::RigidBodyRequests::ApplyAngularImpulse, transformedTorqueVector);
+        }
+    }
+
+    void SimplifiedDriveModel::ApplySpeed(float speed, const ChassisConfiguration& vehicleChassis, uint64_t nsDt)
+    {
+        auto wheelEntities = VehicleDynamics::Utilities::GetAllDriveWheelEntities(vehicleChassis);
+        for (auto& wheelEntity : wheelEntities)
+        {
+            AZ::Transform wheelTransform;
+            AZ::TransformBus::EventResult(wheelTransform, wheelEntity, &AZ::TransformBus::Events::GetWorldTM);
+
+            AZ::Transform inverseWheelTransform = wheelTransform.GetInverse();
+            AZ::Vector3 currentAngularVelocity;
+            Physics::RigidBodyRequestBus::EventResult(currentAngularVelocity, wheelEntity, &Physics::RigidBodyRequests::GetAngularVelocity);
+            currentAngularVelocity = inverseWheelTransform.TransformVector(currentAngularVelocity);
+            auto currentAngularSpeedX = currentAngularVelocity.GetX();
```

Comment:
I think it should be X. I think that we should add constexpr on axis of wheel to measure velocity and apply it correctly.

Reply:
Especially concerning the fact, that each wheel may be rotated differently (which is actually the case in the physics test scene)

Reply:
In [ee4620e](https://github.com/RobotecAI/o3de-ros2-gem/pull/162/commits/ee4620e43bd72ee7184540f630e96374ccfc1d9d) I added temporarily the direction.
In other words it is not hardcoded anymore. The ideal solution would be to find this axis from coordinate frames of joints.
If we reach the ideal solution, `m_drive_dir` and `m_steering_dir` won't be exposed to the user.